lib/service/Service.ts

```diff
@@ -316,6 +325,32 @@ class Service extends EventEmitter {
     return this.pool.listPeers();
   }
 
+  /**
+   * Get all completed swap deals.
+   * @returns A list of completed swap deals.
+   */
+  public listSwaps = async (args: { limit: number, status: ListSwapsRequest.Status }): Promise<SwapDealInstance[]> => {
+    const { limit, status } = args;
+
+    let deals = limit > 0 ?
+      await this.swaps.getCompletedDeals(limit) : await this.swaps.getCompletedDeals();
+
+    switch (status) {
```

Comment:
It would be better to pass in the criteria for whether to return completed, failed, or both kinds of deals to the db query above rather than fetching everything and filtering it in node, which will occupy the node thread longer than needed.